x/auth/client/txbuilder/txbuilder.go

```diff
@@ -212,31 +214,17 @@ func (bldr TxBuilder) BuildAndSign(name, passphrase string, msgs []sdk.Msg) ([]b
 	return bldr.Sign(name, passphrase, msg)
 }
 
-// BuildWithPubKey builds a single message to be signed from a TxBuilder given a set of
-// messages and attach the public key associated to the given name.
-// It returns an error if a fee is supplied but cannot be parsed or the key cannot be
-// retrieved.
-func (bldr TxBuilder) BuildWithPubKey(name string, msgs []sdk.Msg) ([]byte, error) {
-	msg, err := bldr.Build(msgs)
-	if err != nil {
-		return nil, err
-	}
-
-	keybase, err := keys.GetKeyBase()
+// BuildTxForSim creates a StdSignMsg and encodes a transaction with the
+// StdSignMsg with a single empty StdSignature for tx simulation.
+func (bldr TxBuilder) BuildTxForSim(msgs []sdk.Msg) ([]byte, error) {
```

Comment:
This is pretty much the main change -- instead of needing a Keybase, we can just provide an "empty" StdSignature. It'll still be valid and go through all the ante handler steps (for simulation).

Reply:
We don't use it for multi-msg afaik -- correct me if I'm wrong or point me to where said usage is?

Reply:
You're right - sorry for the noise